core/arch/arm/kernel/boot.c

```diff
@@ -381,12 +381,8 @@ static TEE_Result mmap_clear_memtag(struct tee_mmap_region *map,
 				    void *ptr __unused)
 {
 	switch (map->type) {
-	case MEM_AREA_TEE_RAM:
-	case MEM_AREA_TEE_RAM_RW:
 	case MEM_AREA_NEX_RAM_RO:
-	case MEM_AREA_NEX_RAM_RW:
-	case MEM_AREA_TEE_ASAN:
-	case MEM_AREA_TA_RAM:
+	case MEM_AREA_SEC_RAM_OVERALL:
```

Comment:
> Asan memory is still referenced with `MEM_AREA_TEE_ASAN` memory area ID when `CFG_WITH_PAGER` is enabled. Won't this switch case need to be supported?

```
mk/config.mk:999: *** CFG_CORE_SANITIZE_KADDRESS and CFG_MEMTAG are not compatible.  Stop.
```
I don't think we need to care about `MEM_AREA_TEE_ASAN` since it's not used with memory tagging enabled.
 
> `MEM_AREA_TEE_RAM` is still used when `CFG_CORE_RWDATA_NOEXEC` is disabled. IS that an issue? Maybe it's a bit inconsistent to enable `CFG_MEMTAG` but not `CFG_CORE_RWDATA_NOEXEC`.

That shouldn't be a problem, everything should be covered by `MEM_AREA_SEC_RAM_OVERALL` nonetheless.

> Is is OK here to no more clear memtags for `MEM_AREA_TEE_RAM_RW` and `MEM_AREA_NEX_RAM_RW` areas?

The same here, is covered by `MEM_AREA_SEC_RAM_OVERALL`.

However, while looking at logs etc to double-check I noticed:
```
D/TC:0   dump_mmap_table:923 type SEC_RAM_OVERALL va 0x12100000..0x122fffff pa 0x0e100000..0x0e2fffff size 0x00200000 (pgdir)
...
D/TC:0   mmap_clear_memtag:392 Clearing tags for VA 0x12000000..0x121fffff
```

We may have this error in more than one place, see commit f01690c36528 ("core: fix mapping init debug trace") for the `dump_mmap_table()` print. I'll investigate further.